models/purchase_order_enhancement.py

```diff
@@ -0,0 +1,30 @@
+from odoo import api, models, fields, _
+import datetime as dt
+from odoo.exceptions import ValidationError, UserError
+
+class PurchaseOrderEnhancement(models.Model):
+    _inherit = 'purchase.order'
+
+    active = fields.Boolean(default=True)
+    USPhoneNumber = fields.Char(string="Phone Number")
+    lifespan = fields.Integer(string="Lifespan")
+
+    @api.model
+    def schelduled_archive(self):
+        for record in self.search([('state','in',['done', 'cancel'])]):
+            if dt.datetime.now() > record.write_date + dt.timedelta(days=record.lifespan):

```

Comment:
Please combine the` lifespan` condition to search domain instead of performing another check